gossip/src/contact_info.rs

```diff
@@ -0,0 +1,293 @@
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#![allow(unused_imports)]
+pub use crate::legacy_contact_info::LegacyContactInfo;
+use {
+    crate::crds_value::MAX_WALLCLOCK,
+    serde::{Deserialize, Deserializer, Serialize, Serializer},
+    solana_sdk::{
+        pubkey::Pubkey,
+        sanitize::{Sanitize, SanitizeError},
+        serde_varint, short_vec,
+    },
+    std::{
+        borrow::Cow,
+        collections::HashMap,
+        net::{IpAddr, SocketAddr},
+        num::TryFromIntError,
+    },
+    thiserror::Error,
+};
+
+const SOCKET_TAG_GOSSIP: u8 = 0;
+const SOCKET_TAG_REPAIR: u8 = 1;
+const SOCKET_TAG_RPC: u8 = 2;
+const SOCKET_TAG_RPC_PUBSUB: u8 = 3;
+const SOCKET_TAG_SERVE_REPAIR: u8 = 4;
+const SOCKET_TAG_TPU: u8 = 5;
+const SOCKET_TAG_TPU_QUIC: u8 = 6;
+const SOCKET_TAG_TPU_FORWARDS: u8 = 7;
+const SOCKET_TAG_TPU_FORWARDS_QUIC: u8 = 8;
+const SOCKET_TAG_TPU_VOTE: u8 = 9;
+const SOCKET_TAG_TVU: u8 = 10;
+const SOCKET_TAG_TVU_FORWARDS: u8 = 11;
```

Comment:
The opaque `u8` here is intentional so that adding new sockets is backward compatible. 

An `enum` will fail deserialization if there is an unknown tag/enum-variant; as a result you cannot add a new socket in a backward compatible way because nodes not recognizing the enum-variant would fail deserializing the whole contact-info.

You can work around that by adding an `Unknown(u8)` variant to the enum but that would also require manual serde implementation which would be counterproductive.